src/common/perf_counter_connection.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "common/perf_counters.h"
+#include "common/Mutex.h"
+
+class CephContext;
+/* PerfcounterCollection hold the lock for PerfCounterCollectionImpl */
+class PerfCountersCollection
+{
+  CephContext *m_cct;
+
+  /** Protects pf_collection->m_loggers */
+  mutable Mutex m_lock;
+  PerfCountersCollectionImpl *pf_collection;
+public:
+  PerfCountersCollection(CephContext *cct);
+  ~PerfCountersCollection();
+  void add(class PerfCounters *l);
```

Comment:
no need to put `class` here. unless we need to disambiguate the class name and variable name.